pkgs/tools/system/btop/default.nix

```diff
@@ -24,17 +27,24 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ] ++ lib.optionals cudaSupport [
     cudaPackages.autoAddOpenGLRunpathHook
-  ];
+  ] ++ lib.optionals rocmSupport [ makeWrapper ];
 
   buildInputs = lib.optionals stdenv.isDarwin [
     darwin.apple_sdk_11_0.frameworks.CoreFoundation
     darwin.apple_sdk_11_0.frameworks.IOKit
   ];
 
+  runtimeLibs = lib.optionals rocmSupport [
+    rocmPackages.rocm-smi
+  ];
+
   installFlags = [ "PREFIX=$(out)" ];
 
   postInstall = ''
     ${removeReferencesTo}/bin/remove-references-to -t ${stdenv.cc.cc} $(readlink -f $out/bin/btop)
+  '' +
+  lib.optionalString (runtimeLibs != [ ]) ''
+    wrapProgram $(readlink -f $out/bin/btop) --prefix LD_LIBRARY_PATH ':' ${lib.makeLibraryPath runtimeLibs}
```

Comment:
Isn't `btop` an elf binary? Then you can just `patchelf --add-rpath` instead. You'd have to do that in `postPhases`.

P.S. Generally it's best not to rely on `rec` in `mkDerivation rec { ... }` because it doesn't play nice with the overrides. You can use `mkDerivation (finalAttrs: { ... })` or you can lower the logic to the level of bash (`$runtimeLibs` at build time are consistent with the overrides)

Reply:
Like this?

Reply:
Yes thanks!